web/build.sh

```diff
@@ -32,9 +32,9 @@ THE SOFTWARE.
 for the JavaScript code in this page.
 */'
 
-esbuild js/main.mjs --sourcemap --bundle --minify --outfile=static/js/main.mjs "--banner:js=${LICENSE}"
+npx esbuild js/main.mjs --sourcemap --bundle --minify --outfile=static/js/main.mjs "--banner:js=${LICENSE}"
```

Comment:
```suggestion
esbuild js/main.mjs --sourcemap --bundle --minify --outfile=static/js/main.mjs "--banner:js=${LICENSE}"
```

The reason the build scripts are run inside NPM build commands is so that you don't need to use `npx`, which is honestly a massive security problem waiting to happen. This won't work inside the packaging environments that most distros use.

Reply:
Maybe the answer is to append the PATH variable in the makefile to include `node_modules/.bin` as the _last_ element in $PATH?

Reply:
But they are not run in the NPM build commands, they are run in the bash scripts.

Reply:
Yeah but when you run a command from a script via `npm run`, it injects the node_modules bin folder in the PATH:

```json
"env": "env | grep ^PATH="
```

```console
$ npm run env

> @techaro/anubis@1.0.0-see-VERSION-file env
> env | grep ^PATH=

PATH=/home/xe/code/TecharoHQ/anubis/node_modules/.bin:/home/xe/code/TecharoHQ/node_modules/.bin:/home/xe/code/node_modules/.bin:/home/xe/node_modules/.bin:/home/node_modules/.bin:/node_modules/.bin:/home/linuxbrew/.linuxbrew/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/node-gyp-bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/xe/.local/bin:/home/xe/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/xe/go/bin
```

This means that `postcss` and/or `esbuild` just show up in $PATH and work like you installed them as OS-level dependencies.

Reply:
Found another way.